packages/clay-shared/src/useArrowKeysForList.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {useEffect} from 'react';
+
+const UP_ARROW_KEY_CODE = 38;
+const DOWN_ARROW_KEY_CODE = 40;
+
+/**
+ * Hook for handling arrow keys and setting an active index
+ */
+export function useArrowKeysForList(
+	activeIndex: number,
+	totalItems: number,
+	onIndexChange: (newVal: number) => void,
+	element: any = window
+) {
+	useEffect(() => {
+		const handleArrowKeys = (event: KeyboardEvent) => {
+			if (
+				event.keyCode === UP_ARROW_KEY_CODE ||
+				event.keyCode === DOWN_ARROW_KEY_CODE
+			) {
+				event.preventDefault();
+
+				const nextIndex =
+					activeIndex === totalItems - 1 ? 0 : activeIndex + 1;
+				const prevIndex =
+					activeIndex === 0 ? totalItems - 1 : activeIndex - 1;
+
+				onIndexChange(
```

Comment:
We'll probably have to focus on the element too so that the dropdown scroll can track the active item.